spec/unit/http/http_client_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rspec'
+
+module LibyuiClient
+  module Http
+    RSpec.describe HttpClient do
+      let(:http) { Net::HTTPSuccess.new('1.1', 200, 'OK') }
+      before { allow(http).to receive(:body).and_return('{"id": "foo"}') }
+      describe '#http_get' do
```

Comment:
here I would apply something like what he have discussed, for example to avoid that a developer would mess up with internal implementation of this method passing other parameter to `get_response` or if the function would return a string instead of the response it would be detected:
```ruby
  let(:uri) { URI.escape("http://localhost/widgets") }
  let(:response) { Net::HTTPResponse.new('1.1', 200, "OK") }

  before { allow(response).to receive(:body).and_return('{"id": "foo"}') }
  describe '#http_get' do
    it 'calls get_response and return that response' do
      expect(Net::HTTP).to receive(:get_response).with(uri).and_return(response)
      res = HttpClient::http_get(uri)
      expect(res).to eq(response)
    end
  end
```